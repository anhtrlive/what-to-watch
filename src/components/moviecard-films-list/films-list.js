import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {MovieCard} from '../small-movie-card/small-movie-card';
import {withActivePlayer} from '../hocs/moviecard-with-player';

const MovieCardWithPlayer = withActivePlayer(MovieCard);
class FilmsList extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    let {movies, genre} = this.props;
    return <div className="catalog__movies-list">
      {movies.map((filmData, index) => {
        return <MovieCardWithPlayer
          key = {(genre + filmData.id + index).toString()}
          movieName = {filmData.name}
          img = {filmData.previewImage}
          genre = {genre}
          previewVideoLink = {filmData.previewVideoLink}
        />;
      })}
    </div>;
  }
}

FilmsList.propTypes = {
  movies: PropTypes.array.isRequired,
  genre: PropTypes.string,
};

export {FilmsList};
