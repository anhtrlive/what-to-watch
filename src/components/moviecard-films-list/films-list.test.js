import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter} from 'react-router-dom';
import {FilmsList} from './films-list';
import {films} from '../../mocks/films';

Enzyme.configure({adapter: new Adapter()});

const genre = films[0].genre;
const movies = films;

it(`button plus list`, () => {
  const tree = shallow(
      <BrowserRouter>
        <FilmsList
          genre={genre}
          movies={movies}
        />
      </BrowserRouter>).dive();

  expect(tree).toMatchSnapshot();
});
