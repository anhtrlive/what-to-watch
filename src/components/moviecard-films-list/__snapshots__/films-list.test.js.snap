// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button plus list 1`] = `ShallowWrapper {}`;
